### PR TITLE
Comparator extension methods (thenBy, thenByDescending).

### PR DESCRIPTION
**Pull request by davidmartos96**
Issue #108

### CODE REVIEW

**Comment by davidmartos96** [comment]
@passsy Should I ignore the GitHub actions failing? It complains about code unrelated to the PR.

**Comment by passsy** [comment]
You can ignore the missing upload token for codecov. But you have to fix the analysis issues.

Anyways, thank you so much for the PR! Love the feature and I appreciate you added tests 👍

**Comment by davidmartos96** [comment]
@passsy I see that the analyzer errors have been fixed in the branch feature/0.8.0 2863fca41efb15ac7fcc3d81284d82e7cc469fd5 and 99b2639db67058128c98b7237fe6ee8f626a079c
Should I rebase the PR, or keep it for master?

**Comment by passsy** [comment]
Ok, nothing to do from your side then. I'll cherry-pick it to `0.8.0`. Or merge it after merging `0.8.0`. Haven't decided but it will end-up in `master` soon.
`0.8.0` is scheduled for next week.

**Comment by davidmartos96** [comment]
Actually, I think we could add a couple of methods. `thenWith` and `thenWithDescending` which would receive a compartator function instead of a selector. What do you think? @passsy

**Comment by passsy** [comment]
I agree

File: lib/src/collection/comparisons.dart [hunk 1]
```diff
@@ -25,3 +25,28 @@ Comparator<T> compareByDescending<T>(Comparable Function(T) selector) {
   int compareTo(T a, T b) => selector(b).compareTo(selector(a));
   return compareTo;
 }
+
+extension KtComparatorExtensions<T> on Comparator<T> {
+  Comparator<T> thenWith(Comparator<T> comparator) {
```
Comment (passsy) [inline lib/src/collection/comparisons.dart:30]: Add documentation, maybe even an example. (I usually copied what jetbrain already wrote)

File: test/collection/list_mutable_extensions_test.dart [hunk 2]
```diff
@@ -152,6 +152,60 @@ void testList(
...
+          result1,
+          KtList.from(const [
+            KtPair(1, "a"),
+            KtPair(1, "b"),
+            KtPair(2, "c"),
+            KtPair(2, "c"),
+            KtPair(3, "a")
+          ]));
+
+      final result2 =
```
Comment (passsy) [inline test/collection/list_mutable_extensions_test.dart:179]: Split the test cases into multiple tests

File: lib/src/collection/comparisons.dart (hunk 1 above)
Comment (davidmartos96) [reply lib/src/collection/comparisons.dart]: I added some documentation

**Comment by passsy** [comment]
Thank you!

**Comment by davidmartos96** [comment]
@passsy No problem! Awesome project :wink:

**Pull request merged**